src/test/javascript/spec/component/exam/manage/exam-students.component.spec.ts

```diff
@@ -0,0 +1,146 @@
+import * as chai from 'chai';
+import * as sinonChai from 'sinon-chai';
+import * as sinon from 'sinon';
+import { ArtemisTestModule } from '../../../test.module';
+import { ActivatedRoute, convertToParamMap, UrlSegment } from '@angular/router';
+import { Course } from 'app/entities/course.model';
+import { Exam } from 'app/entities/exam.model';
+import { ExamManagementService } from 'app/exam/manage/exam-management.service';
+import { ExamStudentsComponent } from 'app/exam/manage/students/exam-students.component';
+import { StudentsExamImportButtonComponent } from 'app/exam/manage/students/students-exam-import-dialog/students-exam-import-button.component';
+import { MockComponent, MockDirective } from 'ng-mocks';
+import { AlertComponent } from 'app/shared/alert/alert.component';
+import { DataTableComponent } from 'app/shared/data-table/data-table.component';
+import { NgxDatatableModule } from '@swimlane/ngx-datatable';
+import { JhiTranslateDirective } from 'ng-jhipster';
+import { RouterTestingModule } from '@angular/router/testing';
+import { DeleteButtonDirective } from 'app/shared/delete-dialog/delete-button.directive';
+import { User } from 'app/core/user/user.model';
+import { HttpResponse } from '@angular/common/http';
+import { UserService } from 'app/core/user/user.service';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('ExamStudentsComponent', () => {
+    const examCourse = { id: 1 } as Course;
+    const user1 = { id: 1 } as User;
+    const user2 = { id: 2, login: 'user2' } as User;
+
+    const route = ({
+        snapshot: { paramMap: convertToParamMap({ courseId: examCourse.id }) },
+        url: new Observable<UrlSegment[]>(),
+        data: { subscribe: (fn: (value: any) => void) => fn({ exam }) },
+    } as any) as ActivatedRoute;
+
+    let component: ExamStudentsComponent;
+    let fixture: ComponentFixture<ExamStudentsComponent>;
+    let examManagementService: ExamManagementService;
+    let userService: UserService;
+    let exam: Exam;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, NgxDatatableModule, RouterTestingModule],
+            declarations: [
+                ExamStudentsComponent,
+                MockComponent(StudentsExamImportButtonComponent),
+                MockComponent(DataTableComponent),
+                MockComponent(AlertComponent),
+                MockDirective(JhiTranslateDirective),
+                MockDirective(DeleteButtonDirective),
+            ],
+            providers: [{ provide: ActivatedRoute, useValue: route }],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(ExamStudentsComponent);
+        component = fixture.componentInstance;
+        examManagementService = TestBed.inject(ExamManagementService);
+        userService = TestBed.inject(UserService);
+
+        exam = { course: examCourse, id: 2, registeredUsers: [user1, user2] } as Exam;
```

Comment:
Can't you simply move this line to line 42?

Reply:
would probably prevent creating the object every time again, yeah.

Reply:
I cannot since the `registeredUsers` array would persist between tests.